spec/unit/pdk/cli/console_spec.rb

```diff
@@ -0,0 +1,174 @@
+      allow(command).to receive(:execute!).and_return(exit_code: 0)
+      allow(PDK::CLI::Exec::InteractiveCommand).to receive(:new).and_return(command)
+      expect { console_cmd.run(['--puppet-version=5', '--run-once', '--quiet', '--execute=$foo = 123']) }.to exit_zero
+    end
+
+    it 'can pass --pe-version' do
+      allow(PDK::CLI::Util).to receive(:puppet_from_opts_or_env).with({ :"pe-version" => '2018.1' }, true)
+                                                                .and_return(gemset: { puppet: '5.5.10' }, ruby_version: '2.5.3')
+      allow(PDK::Util::Bundler).to receive(:ensure_bundle!).with(puppet: '5.5.10')
+      command = instance_double(PDK::CLI::Exec::InteractiveCommand)
+      allow(PDK::Util::RubyVersion).to receive(:versions).and_return('2.5.3' => '2.5.0')
+      allow(command).to receive(:context=).with(:pwd)
+      allow(command).to receive(:update_environment).with('PUPPET_GEM_VERSION' => '5.5.10')
+      allow(command).to receive(:execute!).and_return(exit_code: 0)
+      allow(PDK::CLI::Exec::InteractiveCommand).to receive(:new).and_return(command)
+      expect { console_cmd.run(['--pe-version=2018.1', '--run-once', '--quiet', '--execute=$foo = 123']) }.to exit_zero
+    end
+
+    it 'can pass --puppet-dev' do
+      allow(PDK::CLI::Util).to receive(:puppet_from_opts_or_env).with({ :"puppet-dev" => '' }, true)
+                                                                .and_return(gemset: { puppet: 'file:///home/user1/.pdk/cache/src/puppet' }, ruby_version: '2.5.3')
+      allow(PDK::Util::Bundler).to receive(:ensure_bundle!).with(puppet: 'file:///home/user1/.pdk/cache/src/puppet')
+      command = instance_double(PDK::CLI::Exec::InteractiveCommand)
+      allow(PDK::Util::RubyVersion).to receive(:versions).and_return('2.5.3' => '2.5.0')
+      allow(command).to receive(:context=).with(:pwd)
+      allow(command).to receive(:update_environment).with('PUPPET_GEM_VERSION' => 'file:///home/user1/.pdk/cache/src/puppet')
+      allow(command).to receive(:execute!).and_return(exit_code: 0)
+      allow(PDK::CLI::Exec::InteractiveCommand).to receive(:new).and_return(command)
+      expect { console_cmd.run(['--puppet-dev', '--run-once', '--quiet', '--execute=$foo = 123']) }.to exit_zero
+    end
+
+    it 'can pass debugger options' do
+      allow(PDK::Util::Bundler).to receive(:ensure_bundle!).with(puppet: '5.4.0')
+      command = instance_double(PDK::CLI::Exec::InteractiveCommand)
+      allow(command).to receive(:context=).with(:pwd)
+      allow(command).to receive(:update_environment).with('PUPPET_GEM_VERSION' => '5.4.0')
+      allow(command).to receive(:execute!).and_return(exit_code: 0)
+      allow(PDK::Util).to receive(:module_root).and_return('/modules/ntp')
+      args = [
+        PDK::CLI::Exec.bundle_bin, 'exec', 'puppet', 'debugger', '--run-once', '--quiet', '--execute=$foo = 123',
+        '--basemodulepath=/modules/ntp/spec/fixtures/modules:/modules/ntp/modules',
```

Comment:
Worth testing here with Windows style paths e.g.
`--basemodulepath=C:\modules\ntp\spec\fixtures\modules;C:\modules\ntp\modules`

Reply:
Added a new test to cover the windows bit.  Also added some helpers to identify what platform.